### PR TITLE
Update vendor of container/storage

### PR DESCRIPTION
**Pull request by rhatdan**
Lots of fixes for issues found by podman.

overlay: propagate errors from mountProgram
utils: root in a userns uses global conf file
Fix handling of additional stores
Correctly check permissions on rootless directory
Fix possible integer overflow on 32bit builds
Evaluate device path for lvm
lockfile test: make concurrent RW test determinisitc
lockfile test: make concurrent read tests deterministic

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@mheon @baude @giuseppe @TomSweeneyRedHat PTAL

File: vendor.conf [hunk 1]
```diff
@@ -19,7 +19,7 @@ github.com/containers/image v1.5.1
 github.com/vbauerster/mpb v3.3.4
 github.com/mattn/go-isatty v0.0.4
 github.com/VividCortex/ewma v1.1.1
-github.com/containers/storage v1.12.1
+github.com/containers/storage v1.12.2
```
Comment (TomSweeneyRedHat) [inline vendor.conf]: should this be 1.12.3?

**Comment by mheon** [comment]
LGTM

File: vendor.conf (hunk 1 above)
Comment (rhatdan) [reply vendor.conf]: Yes repushed.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat @baude @giuseppe @vrothberg Someone please /lgtm on this one.

**Pull request merged**